gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java

```diff
@@ -69,13 +145,21 @@ public void install() {
                     throw new IllegalStateException("Install can only be called once and should be called inside your application onCreate()");
                 }
 
+                if (this.context == null) {
+                    throw new IllegalStateException("You must supply a valid context");
+                }
+
                 if (StringUtils.isBlank(this.bootstrapUrl)) {
                     throw new IllegalStateException("You must supply a bootstrap url");
                 }
 
                 Gandalf gandalf = createInstance();
 
+                gandalf.context = this.context;
                 gandalf.bootstrapUrl = this.bootstrapUrl;
+                gandalf.versionChecker = new DefaultVersionChecker();
+                gandalf.historyChecker = new DefaultHistoryChecker(this.context);
+                gandalf.gateKeeper = new GateKeeper(this.context, gandalf.versionChecker, gandalf.historyChecker);
```

Comment:
Yeah actually I sort of think all the gandalf ones should be final, with non finals in the Installer, and the gandalf constructor should just take each thing, but just be private. any of them not set in the installer create method either get the default ones or the exceptions for the required ones